lib/icinga/legacytimeperiod.cpp

```diff
@@ -406,13 +406,13 @@ Dictionary::Ptr LegacyTimePeriod::FindRunningSegment(const String& daydef, const
 	time_t tsend, tsiter, tsref;
 	int stride;
 
-	tsref = mktime(reference);
+	tsref = Utility::MkTime(reference);
```

Comment:
`LegacyTimePeriod::FindRunningSegment()` is called from `ScheduledDowntime::FindRunningSegment()` with `reference = Utility::LocalTime(Utility::GetTime())`, i.e. a `struct tm` representation of now. If this is done at say 2:30 on a day where this time exists in both CET and CEST, `reference` and `tsref` may be inconsistent in this function.

I didn't attempt to create a test case from that, but it doesn't sound right that `tsref` usually represents the current time, except sometimes, it may be off by an hour.